File: src/main/java/ch/wisv/choice/exam/service/ExamRepository.java
package ch.wisv.choice.exam.service;

import ch.wisv.choice.course.model.Course;
import ch.wisv.choice.exam.model.Exam;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Collection;

@Repository
public interface ExamRepository extends JpaRepository<Exam, Long>{

    Collection<Exam> findByCourse(Course course);
}

File: src/main/kotlin/ch/wisv/choice/exam/service/ExamRepository.kt
package ch.wisv.choice.exam.service

import ch.wisv.choice.exam.model.Exam
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ExamRepository : JpaRepository<Exam, Long> {

    fun findAllByCourse_Code(code: String): List<Exam>

    fun deleteAllByCourse_Code(code: String)
}
